### PR TITLE
Update other-patients name

### LINKED ISSUE #2458: Update "Other Patients" name on the app drawer.
**Describe the feature request.**
Currently, Other patients items on app drawer, one cannot change the name "Other Patients". However, I would like to suggest some flexibility such that one can define a custom name.

**Additional context**
image

**Acceptance criteria**
One should be able to define a custom display name on the config and reflect on the app.

**Area path**
A list of ordered steps in the app on usage of the feature to support anyone testing it e.g. Code reviewer, QA e.g.
1. Login to the app
2. Open app drawer
3. One should be able to see the name defined.

### PR DESCRIPTION
**Pull request by Lentumunai-Mark**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2458
**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt [hunk 1]
```diff
@@ -212,7 +212,8 @@ constructor(
           registerCountMap = appMainUiState.value.registerCountMap,
           menuClickListener = {
             onEvent(AppMainEvent.TriggerWorkflow(navController = event.navController, navMenu = it))
-          }
+          },
+          title = event.title
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:216]: What's the event title here? Is it coming from configuration?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt [hunk 2]
```diff
@@ -216,7 +219,8 @@ private fun OtherPatientsItem(
     onSideMenuClick(
       AppMainEvent.OpenRegistersBottomSheet(
         registersList = navigationConfiguration.bottomSheetRegisters?.registers,
-        navController = navController
+        navController = navController,
+        title = navigationConfiguration.bottomSheetRegisters?.display!!
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt:223]: This has a potential `NullPointerException` with the `!!` on the `display` property. Make `title` nullable instead. Then apply an if condition for when `title.isNullOrEmpty()` - use default `Other patients` string.

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt [hunk 3]
```diff
@@ -213,7 +213,12 @@ constructor(
           menuClickListener = {
             onEvent(AppMainEvent.TriggerWorkflow(navController = event.navController, navMenu = it))
           },
-          title = event.title
+          title =
+            if (event.title.isNullOrEmpty()) {
+              "Other patients"
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:218]: Extract hard-coded string to `string.xml` resource file

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt [hunk 4]
```diff
@@ -220,7 +220,12 @@ private fun OtherPatientsItem(
       AppMainEvent.OpenRegistersBottomSheet(
         registersList = navigationConfiguration.bottomSheetRegisters?.registers,
         navController = navController,
-        title = navigationConfiguration.bottomSheetRegisters?.display
+        title =
+          if (navigationConfiguration.bottomSheetRegisters?.display.isNullOrEmpty()) {
+            "Other Patients"
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/components/AppDrawer.kt:225]: Extract hard-coded string to `string.xml` resource file

File: android/engine/src/androidTest/java/org/smartregister/fhircore/engine/ui/bottomsheet/RegisterBottomSheetFragmentViewKtTest.kt
Comment (pld) [inline android/engine/src/androidTest/java/org/smartregister/fhircore/engine/ui/bottomsheet/RegisterBottomSheetFragmentViewKtTest.kt:1]: please remove Kt from name

**Review by ellykits — APPROVED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**